test/server/cards/02_SHD/leaders/FinnThisIsARescue.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('Finn, This is a Rescue', function () {
+    integration(function (contextRef) {
+        describe('Finn\'s undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['jedi-lightsaber'],
+                        leader: { card: 'finn#this-is-a-rescue', deployed: false },
+                        groundArena: ['battlefield-marine'],
+                        resources: 5
+                    },
+                    player2: {
+                        hand: ['top-target'],
+                        groundArena: ['wampa'],
```

Comment:
I would add an upgrade to the enemy Wampa and ensure it isn't selectable. Also add a TODO for a friendly upgrade on an enemy unit

Reply:
Note: I left a comment asking just to implement that case outright, it requires the explicit action to play an upgrade out to an opponent's unit but I think it's fine to just do that for now so we can have this test done